org.osgi.test.common/src/main/java/org/osgi/test/common/annotation/config/InjectConfiguration.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.common.annotation.config;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/*
+ * Configure the Config-Admin using the annotation `@InjectConfiguration`
+ * Get or Create an new Configuration Object.
+ * Inject the `org.osgi.service.cm.Configuration` (if used on a Parameter or Field) for further updates or delete of the `org.osgi.service.cm.Configuration`
+ *
+ */
+@Inherited
+@Target({
+	TYPE, METHOD, PARAMETER, FIELD
+})
+@Repeatable(Configurations.class)
+@Retention(RUNTIME)
+@Documented
+
+public @interface InjectConfiguration {
+
+	/**
+	 * The pid of the Configuration.
+	 *
+	 * @return The pid
+	 */
+	String pid();
+
+	/**
+	 * Indicate the name of the Factory-Configuration (not {@code null}). If
+	 * empty - auto-generated UUID will be used.
+	 *
+	 * @return The name of the Factory.
+	 */
+
+	String factory_name() default "";
```

Comment:
I think we've used camelCase elsewhere? We should be consistent.

Reply:
Ok! Like filterArguments in InjectService.